# 3.EDA
Obj: Answer the following questions: 
- What do you spend the most money on?
- Compare your income and expenses
- Which account do you use most often, and for what purpose?

In [1]:
import pandas as pd

df = pd.read_csv('C:\\Users\\Admin\\personal-finance-analysis\\data\\personal_transactions.csv')

# Очистка
df['Date'] = pd.to_datetime(df['Date'])
df['Month']   = df['Date'].dt.to_period('M')
df['Year']    = df['Date'].dt.year
df['Weekday'] = df['Date'].dt.day_name()

df_clean = df[df['Category'] != 'Credit Card Payment'].copy()

expenses = df_clean[df_clean['Transaction Type'] == 'debit'].copy()
income   = df_clean[df_clean['Transaction Type'] == 'credit'].copy()

Q1: What do you spend your money on?

In [3]:
by_category = (expenses.groupby('Category')['Amount'].sum().sort_values(ascending=False))
print(by_category)

Category
Mortgage & Rent           24754.50
Home Improvement          19092.87
Groceries                  2795.21
Utilities                  2776.00
Restaurants                2613.02
Shopping                   1973.24
Gas & Fuel                 1715.17
Mobile Phone               1680.40
Internet                   1570.88
Auto Insurance             1350.00
Electronics & Software      719.00
Alcohol & Bars              539.13
Haircut                     378.00
Fast Food                   330.63
Music                       224.49
Movies & DVDs               222.19
Coffee Shops                115.54
Television                  104.78
Food & Dining                77.75
Entertainment                 9.62
Name: Amount, dtype: float64


In [4]:
avg_by_category = (expenses.groupby('Category')['Amount'].mean().sort_values(ascending=False))
print(avg_by_category)

Category
Mortgage & Rent           1178.785714
Home Improvement           530.357500
Electronics & Software     179.750000
Mobile Phone                80.019048
Auto Insurance              75.000000
Internet                    74.803810
Utilities                   44.063492
Food & Dining               38.875000
Gas & Fuel                  32.984038
Shopping                    32.887333
Restaurants                 32.259506
Haircut                     29.076923
Groceries                   26.621048
Alcohol & Bars              21.565200
Fast Food                   20.664375
Television                  13.097500
Movies & DVDs               12.343889
Music                       10.690000
Entertainment                9.620000
Coffee Shops                 3.727097
Name: Amount, dtype: float64


In [5]:
avg_by_weekday = (expenses.groupby('Weekday')['Amount'].mean().sort_values(ascending=False))
print(avg_by_weekday)

Weekday
Friday       183.960494
Thursday     171.486989
Tuesday      115.313939
Monday        94.291016
Wednesday     56.731899
Sunday        43.596923
Saturday      25.774490
Name: Amount, dtype: float64


So, to answer the first question, we can say that $24,754.50 was spent on Mortgage & Rent, this category was also the most expensive on average, and the most money was spent on average on Friday.

Q2: Compare income and expenses

In [7]:
monthly_expenses = expenses.groupby('Month')['Amount'].sum()
monthly_income = income.groupby('Month')['Amount'].sum()

balance = monthly_income - monthly_expenses

print("Monthly expenses:")
print(monthly_expenses)


Monthly expenses:
Month
2018-01     2066.65
2018-02     2089.44
2018-03     2178.66
2018-04     2862.66
2018-05    10507.56
2018-06     2115.05
2018-07     2302.64
2018-08     1992.68
2018-09     2314.74
2018-10     2242.37
2018-11     2118.15
2018-12     2652.94
2019-01     1736.43
2019-02     1954.60
2019-03     2142.59
2019-04     1966.87
2019-05     2253.41
2019-06    10912.38
2019-07     2551.12
2019-08     2043.20
2019-09     2038.28
Freq: M, Name: Amount, dtype: float64


In [8]:
print("Monthly income:")
print(monthly_income)

Monthly income:
Month
2018-01    4000.0
2018-02    4000.0
2018-03    6000.0
2018-04    4000.0
2018-05    4000.0
2018-06    4000.0
2018-07    4000.0
2018-08    6000.0
2018-09    4000.0
2018-10    4000.0
2018-11    4000.0
2018-12    4000.0
2019-01    4000.0
2019-02    4000.0
2019-03    6000.0
2019-04    4000.0
2019-05    4000.0
2019-06    4000.0
2019-07    4500.0
2019-08    6750.0
2019-09    4500.0
Freq: M, Name: Amount, dtype: float64


In [9]:
print("Monthly balance:")
print(balance)

Monthly balance:
Month
2018-01    1933.35
2018-02    1910.56
2018-03    3821.34
2018-04    1137.34
2018-05   -6507.56
2018-06    1884.95
2018-07    1697.36
2018-08    4007.32
2018-09    1685.26
2018-10    1757.63
2018-11    1881.85
2018-12    1347.06
2019-01    2263.57
2019-02    2045.40
2019-03    3857.41
2019-04    2033.13
2019-05    1746.59
2019-06   -6912.38
2019-07    1948.88
2019-08    4706.80
2019-09    2461.72
Freq: M, Name: Amount, dtype: float64


In [11]:
negative_months = balance[balance < 0]
positive_months = balance[balance > 0]

print("Months with negative balance:")
print(negative_months)
print("Months with positive balance:")
print(positive_months)
print(f"Worst month: {balance.idxmin()} with balance {balance.min():.2f}")
print(f"Best month: {balance.idxmax()} with balance {balance.max():.2f}")


Months with negative balance:
Month
2018-05   -6507.56
2019-06   -6912.38
Freq: M, Name: Amount, dtype: float64
Months with positive balance:
Month
2018-01    1933.35
2018-02    1910.56
2018-03    3821.34
2018-04    1137.34
2018-06    1884.95
2018-07    1697.36
2018-08    4007.32
2018-09    1685.26
2018-10    1757.63
2018-11    1881.85
2018-12    1347.06
2019-01    2263.57
2019-02    2045.40
2019-03    3857.41
2019-04    2033.13
2019-05    1746.59
2019-07    1948.88
2019-08    4706.80
2019-09    2461.72
Freq: M, Name: Amount, dtype: float64
Worst month: 2019-06 with balance -6912.38
Best month: 2019-08 with balance 4706.80


Over 21 months of observation, 19 months were profitable, and only two were in the red: May 2018 (-$6,507) and June 2019 (-$6,912). This indicates that the budget is generally managed stably, although these two months are outliers.

Q3:Which account do you use most often, and for what purpose?

In [13]:
by_account_count = (expenses.groupby('Account Name')['Amount'].count().sort_values(ascending=False))

print("Number of transactions :")
print(by_account_count)

Number of transactions :
Account Name
Platinum Card    324
Checking         147
Silver Card      146
Name: Amount, dtype: int64


In [14]:
by_account_sum = (expenses.groupby('Account Name')['Amount'].sum().sort_values(ascending=False))

print("Total amount spent :")
print(by_account_sum)

Total amount spent :
Account Name
Checking         49456.78
Platinum Card     8996.31
Silver Card       4589.33
Name: Amount, dtype: float64


In [17]:
by_account_category = (expenses.groupby(['Account Name', 'Category'])['Amount'].sum().sort_values(ascending=False))

print("Amount spent by account and category:")
print(by_account_category)

Amount spent by account and category:
Account Name   Category              
Checking       Mortgage & Rent           24754.50
               Home Improvement          17325.00
               Utilities                  2776.00
Platinum Card  Shopping                   1973.24
Checking       Mobile Phone               1680.40
               Internet                   1570.88
Platinum Card  Groceries                  1489.28
Checking       Auto Insurance             1350.00
Silver Card    Restaurants                1313.33
               Groceries                  1305.93
Platinum Card  Restaurants                1299.69
               Home Improvement           1114.67
               Gas & Fuel                  947.02
Silver Card    Gas & Fuel                  768.15
               Home Improvement            653.20
Platinum Card  Electronics & Software      652.68
               Alcohol & Bars              354.83
               Fast Food                   315.40
               Haircut  

Each account plays a specific role in the budget:

**Checking Account** — used exclusively for major fixed payments: mortgage ($24,754), utilities ($2,776), mobile service ($1,680), and internet ($1,570). These are fixed expenses that remain consistent month to month.

**Platinum Card** — the primary card for everyday spending: shopping ($1,973), groceries ($1,489), dining out ($1,299), and fuel ($947). It is also used for entertainment, subscriptions, and coffee.

**Silver Card** — used alongside the Platinum card, mainly for dining out ($1,313) and groceries ($1,305). It essentially duplicates the Platinum Card’s role in the food and dining category.

## Final Observations
### 1. Two main expenses dominate the entire budget
Mortgage ($24,754) and Home Improvement ($19,092) together account for about 72% of all expenses over the 21-month period. All other categories — such as groceries, dining out, and transport — pale in comparison. When planning the budget, these two items should be considered separately from daily spending.
### 2. The budget is stable, with two anomalous months
19 out of 21 months ended with a surplus of approximately $1,700–$2,000. May 2018 (-$6,507) and June 2019 (-$6,912) are sharp outliers — likely due to major Home Improvement costs, where the average transaction is $530.
### 3. Food is a hidden major expense category
Combining groceries, restaurants, coffee, and fast food results in a total of $5,931, making it the third-largest expense after the mortgage and home repairs. Since food spending is split between the Platinum and Silver cards, it is more difficult to track and control.
### 4. Each account serves a specific role

* Checking: Used strictly for mandatory fixed payments.
* Platinum Card: The primary card for daily expenses.
* Silver Card: Duplicates the Platinum Card in food and dining categories without a clear reason for the split.

### 5. Friday and Thursday are the "most expensive" days of the week
The average transaction on Friday ($183) is three times higher than on Saturday ($25). This is counterintuitive, as one might expect spending to peak over the weekend. It is likely that large scheduled payments occur at the end of the work week.

